CommMicro.py

```diff
@@ -65,86 +65,113 @@ def getPath(fileName):
+        number_of_channels = self.counter
+        self.ui.label_samplingrate.setNum(sampling_rate)
+        self.ui.label_acq_time.setNum(
+            BUFFER_LENGTH * decimation * number_of_buffers / (sampling_rate * number_of_channels))
+
+    @Slot(int)
+    def update_counter(self, state):
+        if state == Qt.Unchecked:
+            self.counter -= 1
+        elif state == Qt.Checked:
+            self.counter += 1
+        self.update_daq_setting()
+
     def ChangeCav(self):
-#   This function responds to a user changing the cavity combo box
-#    from cavs 1-4 to cavs 5-8
-        cavs=self.ui.CavComboBox.currentIndex()
-        if cavs==0:
-            delta=1
+        #   This function responds to a user changing the cavity combo box
+        #    from cavs 1-4 to cavs 5-8
+        cavs = self.ui.CavComboBox.currentIndex()
+        if cavs == 0:
+            delta = 1
         else:
-            delta=5
-        for idx,cb in enumerate(self.checkboxes):
-            cb.setText(str(idx+delta))
-
-
-# This doesn't work yet
-# Function to print the window
-#    def plotWindow(self):
-#        fname='plot.png'
-#        app = QtWidgets.QApplication(sys.argv)
-#        screen = QtWidgets.QApplication.primaryScreen()
-#        screenshot = screen.grabWindow()
-#        QScreen.grabWindow(app.primaryScreen(),
-#          QApplication.desktop().winId()).save(fname,'png')
-#        if path.exists(fname) and path.getsize(fname):
-#            try:
-#                system('lpr -Pphysics-lcls2log '+fname)
-#            except:
-#                print('Unable to print {} with apologies'.format(fname))
-#        else:
-#            print('creation of {} failed'.format(fname))
-
-# This function takes given data (cavUno) and axis handle (tPlot) and calculates FFT and plots
-    def FFTPlot(self, bPlot,cavUno):   
+            delta = 5
+        for idx, cb in enumerate(self.checkboxes):
+            cb.setText(str(idx + delta))
+
+    # This doesn't work yet
+    # Function to print the window
+    #    def plotWindow(self):
+    #        fname='plot.png'
+    #        app = QtWidgets.QApplication(sys.argv)
+    #        screen = QtWidgets.QApplication.primaryScreen()
+    #        screenshot = screen.grabWindow()
+    #        QScreen.grabWindow(app.primaryScreen(),
+    #          QApplication.desktop().winId()).save(fname,'png')
+    #        if path.exists(fname) and path.getsize(fname):
+    #            try:
+    #                system('lpr -Pphysics-lcls2log '+fname)
+    #            except:
+    #                print('Unable to print {} with apologies'.format(fname))
+    #        else:
+    #            print('creation of {} failed'.format(fname))
+
+    # This function takes given data (cavUno) and axis handle (tPlot) and calculates FFT and plots
+    def FFTPlot(self, bPlot, cavUno):
 
         N = len(cavUno)
-        T = 1.0/1000 
+        T = 1.0 / (DEFAULT_SAMPLING_RATE / int(self.ui.comboBox_decimation.currentText()))

```

Comment:
Can we give these variables more descriptive names lol What is T?

Reply:
made variable names more descriptive